cmd/internal/initializer/initializer.go

```diff
@@ -136,33 +139,45 @@ func (i *Initializer) initialize(ctx context.Context) error {
 		return fmt.Errorf("unable to ensure backup bucket: %w", err)
 	}
 
+	i.log.Info("ensuring default download directory")
+	err = os.MkdirAll(constants.DownloadDir, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to ensure default download directory: %w", err)
+	}
+
 	i.log.Info("checking database")
 	i.currentStatus.Status = v1.StatusResponse_CHECKING
 	i.currentStatus.Message = "checking database"
 
-	needsBackup, err := i.db.Check(ctx)
+	versions, err := i.bp.ListBackups(ctx)
 	if err != nil {
-		return fmt.Errorf("unable to check data of database: %w", err)
+		return fmt.Errorf("unable retrieve backup versions: %w", err)
 	}
 
-	if !needsBackup {
-		i.log.Info("database does not need to be restored")
+	latestBackup := versions.Latest()
+	if latestBackup == nil {
+		i.log.Info("there are no backups available, it's a fresh database. allow database to start")
 		return nil
 	}
 
-	i.log.Info("database potentially needs to be restored, looking for backup")
+	if i.encrypter == nil {
+		if encryption.IsEncrypted(latestBackup.Name) {
+			return fmt.Errorf("latest backup is encrypted, but no encryption/decryption is configured")
+		}
+	}
 
-	versions, err := i.bp.ListBackups(ctx)
+	needsBackup, err := i.db.Check(ctx)
 	if err != nil {
-		return fmt.Errorf("unable retrieve backup versions: %w", err)
+		return fmt.Errorf("unable to check data of database: %w", err)
 	}
 
-	latestBackup := versions.Latest()
-	if latestBackup == nil {
-		i.log.Info("there are no backups available, it's a fresh database. allow database to start")
+	if !needsBackup {
+		i.log.Info("database does not need to be restored")
 		return nil
 	}
 
+	i.log.Info("database potentially needs to be restored, looking for backup")
```

Comment:
In order to increase availability of the DB we should do this check before validating the backups.